keras_nlp/models/gpt2/gpt2_causal_lm.py

```diff
@@ -200,17 +213,138 @@ def backbone_cls(cls):
     def preprocessor_cls(cls):
         return GPT2CausalLMPreprocessor
 
-    def _get_token_probability(self, prompt, mask):
+    def call_with_cache(self, inputs, cache, cache_index=None):
```

Comment:
Let's just move all the inputs into an input dict. So `inputs` a dict of `token_ids`, `padding_mask`, `cache`, `cache_index`.

Having two different levels where arguments are supplied is confusing. We want the mental model to be this is just a separate forward pass with new normal inputs added.